tests/test_core.py

```diff
@@ -36,64 +36,89 @@ def test_inspect_dummy_execs():
 
 
 @pytest.mark.parametrize(
-    'package_path, expected_path',
+    'package_path, expected_path, build_args_envs',
     [
         (
             Path(__file__).parent / 'cases' / 'executor_1',
             Path(__file__).parent / 'cases' / 'executor_1.json',
+            { 
+                'AUTH_TOKEN': "AUTH_TOKEN",
+                'TOKEN': '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'
+            } ,
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_2',
             Path(__file__).parent / 'cases' / 'executor_2.json',
+            {},
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_3',
             Path(__file__).parent / 'cases' / 'executor_3.json',
+            {},
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_4',
             Path(__file__).parent / 'cases' / 'executor_4.json',
+            {},
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_5',
             None,
+            {},
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_6',
             None,
+            {},
+        ),
+        (
+            Path(__file__).parent / 'cases' / 'executor_7',
+            None,
+            { 
+                'AUTH_TOKEN': "AUTH_TOKEN",
+                'TOKEN': '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'
+            } ,
         ),
         (
             Path(__file__).parent / 'cases' / 'nested',
             Path(__file__).parent / 'cases' / 'nested.json',
+            {},
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_2',
             None,
+            { 
+                'AUTH_TOKEN': "AUTH_TOKEN",
+                'TOKEN': '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'
+            } ,
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_3',
             None,
+            {},
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_4',
             None,
+            {},
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_5',
             None,
+            {},
         ),
     ],
 )
-def test_get_executor_args(package_path, expected_path):
+def test_get_executor_args(package_path, expected_path, build_args_envs):
```

Comment:
The test doesn't verify whether the `ARG` is correctly inserted/added to the Dockerfile

Reply:
let me think a little, and improve it